src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -461,170 +538,229 @@ class boss_archimonde : public CreatureScript
+                            break;
                         }
                     }
-                    CheckDistanceTimer = 5000;
-                } else CheckDistanceTimer -= diff;
-            }
-
-            if (BelowTenPercent)
-            {
-                if (!HasProtected)
-                {
+                    events.RepeatEvent(5000);
+                    break;
+                }
+                case EVENT_BELOW_10_PERCENT_HP:
+                    DoCastProtection();     // Protection of Elune against Finger and Hand of Death
+                    BelowTenPercent = true;
+                    me->GetMotionMaster()->Clear(false);
+                    me->GetMotionMaster()->MoveIdle();
+                    events.ScheduleEvent(EVENT_ENRAGE, 0);
+                    events.ScheduleEvent(EVENT_ENRAGE_ROOT, 0);
+                    events.ScheduleEvent(EVENT_SUMMON_WISPS, 1000);
+                    events.ScheduleEvent(EVENT_SPELL_HAND_OF_DEATH, 1500);
+                    events.ScheduleEvent(EVENT_SPELL_FINGER_OF_DEATH, 2500);
+                    break;
+                case EVENT_SUMMON_WISPS:
+                    // If there are more than 30 Wisps then kill Archimonde
+                    if (WispCount >= 30)
+                    {
+                        Unit::DealDamage(me, me, me->GetHealth(), NULL, DIRECT_DAMAGE, SPELL_SCHOOL_MASK_NORMAL, NULL, false);
+                        return;     // Finish the encounter and no more event repeat
+                    }
+                    DoSpawnCreature(CREATURE_ANCIENT_WISP, float(rand() % 40), float(rand() % 40), 0, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000);
+                    ++WispCount;
+                    events.ScheduleEvent(EVENT_SUMMON_WISPS, 1500);
+                    break;
+                case EVENT_SPELL_HAND_OF_DEATH:
+                    DoCastVictim(SPELL_RED_SKY_EFFECT);
+                    DoCastVictim(SPELL_HAND_OF_DEATH);
+                    events.ScheduleEvent(EVENT_SPELL_HAND_OF_DEATH, 3000);
+                    break;
+                case EVENT_SPELL_FINGER_OF_DEATH:
+                    if (CanUseFingerOfDeath())
+                    {
+                        Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0);
+                        DoCast(target, SPELL_FINGER_OF_DEATH);
+                        DoCastVictim(SPELL_RED_SKY_EFFECT);
+                    }
+                    events.ScheduleEvent(EVENT_SPELL_FINGER_OF_DEATH, 3500);
+                    break;
+                case EVENT_SPELL_GRIP_OF_THE_LEGION:
+                    DoCast(SelectTarget(SELECT_TARGET_RANDOM, 0), SPELL_GRIP_OF_THE_LEGION);
+                    events.ScheduleEvent(EVENT_SPELL_GRIP_OF_THE_LEGION, urand(5000, 25000));
+                    break;
+                case EVENT_SPELL_AIR_BURST:
+                    Talk(SAY_AIR_BURST);
+                    DoCast(SelectTarget(SELECT_TARGET_RANDOM, 1), SPELL_AIR_BURST); // Not on tank
```

Comment:
It can be casted on tank 

> At a point in our first attempt, the tank was Air Bursted while no one else was near the boss, causing us quite a silly wipe.

https://www.wowhead.com/npc=17968/archimonde#comments:id=1114086